dotCMS/src/main/java/com/dotcms/util/user/LiferayUserTransformer.java

```diff
@@ -0,0 +1,161 @@
+package com.dotcms.util.user;
+
+import com.dotmarketing.db.DbConnectionFactory;
+import com.liferay.portal.model.User;
+import java.util.Map;
+
+/**
+ * This class implement the UserTransformer interface to convert dotconnect maps from user to User
+ */
+public class LiferayUserTransformer implements UserTransformer {
+
+
+    @Override
+    public User fromMap(final Map<String, Object> map) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 89 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This method has 145 lines, which is greater than the 75 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS138)